woocommerce/woocommerce-ios
# Order Creation: Add fees UI

Closes: https://github.com/woocommerce/woocommerce-ios/issues/6027

## Description

This PR adds navigation to the fully functional "Fee Details" screen from the new order screen.

## Changes

- Adds `FeeLineDetails` view.
- Adds `feesBaseAmountForPercentage` to `PaymentDataViewModel`.
- Connects navigation and data flow to `FeeLineDetails` view.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "Add Product" button and select any product to enable "Payment" section.
5. Tap "Add Fees" button.
6. Check the Fee Details screen, enter a fixed amount and save.
7. Confirm that the "Payment" section displays correct fees and total values.
8. Tap the "Fees" row to open Fee Details, check that it's prefilled correctly.
9. Switch to the percentage fee type, save.
10. Confirm that the "Payment" section displays correct fees and total values.
11. Tap the "Fees" row -> "Delete Fee from Order".
12. Confirm that the fees row is replaced with a button and the "Order Total" is correctly updated.

## Video

https://user-images.githubusercontent.com/3132438/155014512-4286778f-0700-4607-92d5-88acb4b1db6e.mp4

---
